src/service_alerts.rs

```diff
@@ -24,14 +24,40 @@ pub const FINALIZER_NAME: &str = "servicealert.cactuar.rs";
     namespaced
 )]
 pub struct ServiceAlertSpec {
-    pub common_labels: HashMap<String, String>,
+    pub common_labels: CommonLabels,
     pub deployment_name: String,
-    pub alerts: HashMap<Alerts, Vec<AlertConfig>>,
+    pub alerts: Alerts,
+}
+
+// Since the metrics are different for different protocols, we must map each Alerts enum
+// to a different expression string in prometheus land.
+// e.g.
+// REST + ErrorPercent uses the istio_requests_total         istio standard metric
+// gRPC + ErrorPercent uses the istio_request_messages_total istio standard metric
+#[derive(Debug, Serialize, Deserialize, Clone, JsonSchema, PartialEq)]
+pub struct Alerts {
+    #[serde(rename = "gRPC")]
+    pub GRPC: Option<HashMap<HttpAlerts,Vec<HttpAlertConfig>>>,
+    #[serde(rename = "REST")]
+    pub REST: Option<HashMap<HttpAlerts,Vec<HttpAlertConfig>>>,
+    // TODO: Define what is needed for misc here. look into what alerts we can support first
+    // There's a problem that the alertConfig changes for potentially every different misc
+    // alert so we can't support the Vec<HttpAlertConfig> pattern like we do with gRPC and REST
+    // pub Misc: Option<HashMap<MiscAlerts, $something>>,
+}
+
+// It would be cool if we could union this struct with a HashMap<String, String>
+// so that we can validate at deploy time the owner and origin fields but also allow
+// arbitrary fields
+#[derive(Debug, Serialize, Deserialize, Clone, JsonSchema, PartialEq, Eq)]
```

Comment:
see comment

Reply:
Agreed, I think we can accomplish this using `#[serde(flatten)]`: https://serde.rs/attr-flatten.html#capture-additional-fields

Would probably look like this:
```rust
#[derive(Debug, Serialize, Deserialize, Clone, JsonSchema, PartialEq, Eq)]
pub struct CommonLabels {
    pub owner: String,
    pub origin: String,
    #[serde(flatten)]
    pub extra: HashMap<String, String>
}
```

Reply:
Confirmed this works on my branch: 

- https://github.com/harrytucker/cactuar/pull/29/files#diff-9eadfcdffb66913a15d10e5208a4fde8554cc53aa4b3156bd64d6a72097934ceR20-R22
- https://github.com/harrytucker/cactuar/pull/29/files#diff-2678d0c5a7d623949f960781dd74cb621c58835becae7dc6049ba2596fd7da44R29-R35